### ISSUE TITLE
Add the nextflow summary to MultiQC reports

### ISSUE THREAD

**Issue by ewels** [nf-core/tools#92]
_From @ewels on May 24, 2018 15:25_

In other pipelines we are adding the `summary` variables to MultiQC as a config block. See https://github.com/wikiselev/rnaseq/blob/eaebf588e83e2f78cea0a4451db2d4eea5789493/main.nf#L1036-L1054

Add this into the cookiecutter recipe. @pditommaso thinks it _"could be replaced by a few lines of groovy"_ so probably room for some refactoring too 😉

_Copied from original issue: nf-core/cookiecutter#28_

**Comment by ewels** [comment]
_From @pditommaso on May 24, 2018 17:27_

That can be done with plain groovy code: 

```nextflow
def create_summary() {

    def yaml_file = workDir.resolve('workflow_summary_mqc.yaml')
    yaml_file.text  = """
    id: 'nfcore-rnaseq-summary'
    description: " - this information is collected when the pipeline is started."
    section_name: 'nfcore/rnaseq Workflow Summary'
    section_href: 'https://github.com/nf-core/rnaseq'
    plot_type: 'html'
    data: |
        <dl class=\"dl-horizontal\">
${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }.join("\n")}
        </dl>
    """.stripIndent()

   return yaml_file
}


process foo {
  input: 
  file summary from create_summary()
  '''
  your_command --etc $summary
  '''
}
```

**Comment by ewels** [comment]
_From @MaxUlysse on May 28, 2018 9:6_

OMG! That's two processes I can get rid of.

**Comment by ewels** [comment]
Thanks @pditommaso! Definitely cleaner (and hopefully faster) than having that code in a process 👍 

> That's two processes I can get rid of.

How so? There's only one process for the pipeline summary, no? The software version stuff is separate and still needs a process, as the tools need to be executed to get the versions.

**Comment by ewels** [comment]
_From @MaxUlysse on May 28, 2018 13:27_

I'm making the process for getting all the version tools, so I was thinking of maybe trying to get it smarter as well, but it might have been to soon for me to shout `Eureka!`

PS: We do have a saying in French: _"Il ne faut pas vendre la peau de l'ours avant de l'avoir tuer"_ which with a literal translation goes by _"You should not sell the bearskin before you killed it"_, which I guess should have something to do with chicken hatching in English. (I know that @sven1103 likes that stuff).

**Comment by stevekm** [comment]
I see this is listed as "done", but I am not clear what this feature was added to or how to use it. Are there and docs available on it? I found this by searching Google for "add Nextflow workflow summary to Multiqc"

for the time being, I have reverse-engineered a relatively simple equivalent method here https://github.com/stevekm/nextflow-demos/tree/master/workflow-summary

since I am implementing it at the Nextflow / Multiqc level and not actually with nf-core tools, this is likely the wrong repo for this specific request, hopefully someone can guide me to the correct one. Thanks.
